services-geojson/build.gradle

```diff
@@ -1,12 +1,71 @@
-apply plugin: 'java-library'
+import org.jetbrains.kotlin.gradle.plugin.mpp.apple.XCFrameworkConfig
 
-dependencies {
-  // Gson
-  api libs.gson
+plugins {
+    alias(libs.plugins.kotlin.multiplatform)
+    alias(libs.plugins.kotlinx.serialization)
+    id("maven-publish")
+    alias(libs.plugins.kotlin.cocoapods)
+}
+
+kotlin {
+    jvmToolchain(17)
+
+    jvm()
+
+    def xcf = new XCFrameworkConfig(project)
+    [iosX64(), iosArm64(), iosSimulatorArm64()].forEach { target ->
+        target.binaries.framework { framework ->
+            baseName = "maplibre-geojson"
+            xcf.add(framework)
+        }
+    }
+
+    sourceSets {
+        commonMain.dependencies {
+            api libs.kotlinx.serialization.json
+        }
 
-  // Annotations
-  compileOnly libs.supportAnnotation
+        commonTest.dependencies {
+            implementation libs.kotlin.test
+            implementation libs.kotlinx.io.core
+        }
 
-  // Test Dependencies
-  testImplementation libs.okhttp3Mockwebserver
+        jvmMain.dependencies {
+            api libs.gson
+        }
+    }
+
+    cocoapods {
```

Comment:
Should we really use cocoapods vs. SPM? I am not 100% sure what is possible here, but I think it should be possible to use SPM here?

Reply:
Good point, we I think we should support both!

Reply:
Cocoapods is officially supported by jetbrains, but there are some third party plugins for SPM. I haven't used those (but I imagine they should work fine)

Something I'm unsure of: if this becomes a dependency MapLibre iOS via SPM/Cocoapods, then a Kotlin Multiplatform project (MapLibre Compose) depends on MapLibre iOS via cocoapods, what does the build config look like for that Kotlin project? Is it `compileOnly` to this library, expecting Cocoapods / XCode to link it in the end?

Probably something to explore with a prerelease

Reply:
Good, and important point!

We used this already in navigation core (also KMP). This library was included their (as added on gradle with `api`). But the names where converted to something like `Services_geojsonLineString`. 

I found out, that you can [`export` libraries in cocoapods](https://github.com/maplibre/maplibre-java/blob/a92c8e304fd6d2f4c9d038d6a3e21947f2fccc33/services-turf/build.gradle#L49). Then the sub-dependencies can be used with clean names (ex: `LineString`). But I don't know, how it will be handled when multiple libraries using the same dependency. But I think and hope their will already a good solution for this. This needs also to be handled in native iOS libraries (without KMP).

But good to mention, I will have a look at this 👍